examples/inference_model/ada_svr.py

```diff
@@ -0,0 +1,191 @@
+"""
+ADA-SVR: Adaptive Permutation Threshold Support Vector Regression
+==================================================================
+Statistical inference procedure presented in :footcite:t:`gaonkar_deriving_2012`.
+"""
+
+#############################################################################
+# Imports needed for this script
+# ------------------------------
+import matplotlib.pyplot as plt
+import numpy as np
+from hidimstat.ada_svr import ada_svr, ada_svr_pvalue
+from hidimstat.permutation_test import permutation_test
+from sklearn.svm import SVR
+from hidimstat.scenario import multivariate_1D_simulation
+from hidimstat.visualisation.plot_dataset import (
+    plot_dataset1D,
+    plot_validate_variable_importance,
+    plot_pvalue_H0,
+    plot_pvalue_H1,
+    plot_compare_proba_estimated,
+)
+
+#############################################################################
+# Generate toy dataset
+# --------------------
+#
+
+# Parameters for the generation of data
+n_samples, n_features = 20, 100
+support_size = 1
+
+X, y, beta, _ = multivariate_1D_simulation(
+    n_samples=n_samples,
+    n_features=n_features,
+    support_size=support_size,
+    sigma=0.1,
+    shuffle=False,
+    seed=42,
+)
+plot_dataset1D(X=X, y=y, beta=beta)
+#############################################################################
+# Usage the methods
+# -----------------
+# see the API for more details about the optional parameter:
+# :py:func:`hidimstat.ada_svr`
+
+beta_hat, scale = ada_svr(X, y)
+
+#############################################################################
+# | **beta_hat** is the estimated variable of importance
+# | **scale** is the standard deviation of the distribution of the coefficients
+
+#############################################################################
+# Plot the results
+# ----------------
+#
+plot_validate_variable_importance(beta, beta_hat)
+
+#############################################################################
+# The result shows that the only variable of importance has a higher score
+# but the difference between beta and beta_hat is quite different.
+
+#############################################################################
+#
+# The pvalue and the corrected pvalue can help to be confident in the previous
+# results.
+
+
+pvalue, pvalue_corrected, one_minus_pvalue, one_minus_pvalue_correlation = (
+    ada_svr_pvalue(beta_hat, scale)
+)
+plot_pvalue_H0(
```

Comment:
I don't think it's a good idea to have plotting functions as black boxes. See examples from sklearn and nilearn, where this pattern is avoided.

Reply:
In my opinion: 
 - Having predefined functions makes it harder for users to tweak the plotting script to adapt the figure to their constraints (publication template, presentation ... )
 - The plotting functionalities are currently not too heavy; it wouldn't reduce the readability of the notebook to include them. 
 - There is redundancy in plotting coefficients, p-values, and corrected p-values. Unless you want to highlight what one of these quantities adds on top of the others

Reply:
Do you propose to only plot corrected p-values?